### PR TITLE
ws: Fix GOptionContext memory leak

### PR DESCRIPTION
**Pull request by martinpitt**
Free the context after use. This isn't a big issue in practice, but
valgrind complains about it when running cockpit-ws as part of a unit
test.

### CODE REVIEW

File: src/ws/main.c [hunk 1]
```diff
@@ -146,11 +147,10 @@ main (int argc,
 
   context = g_option_context_new (NULL);
   g_option_context_add_main_entries (context, cmd_entries, NULL);
-
-  if (!g_option_context_parse (context, &argc, &argv, error))
-    {
-      goto out;
-    }
+  success = g_option_context_parse (context, &argc, &argv, error);
+  g_option_context_free (context);
```
Comment (allisonkarlitskaya) [inline src/ws/main.c]: Why not just include this under the `out:` block with everything else?

....also, remind me again why we're not using g_autoptr()?

Comment (martinpitt) [reply src/ws/main.c]: I had it in `out:` at first, but there's no need to keep the context allocated throughout ws'es lifetime.

We can use g_autoptr now, but that would be functionally equivalent of free'ing it in `out:`, right?

Comment (allisonkarlitskaya) [reply src/ws/main.c]: This seems like a pretty small optimisation to sacrifice idiomatic readability for....

Comment (martinpitt) [reply src/ws/main.c]: Ack. I moved to g_auto now, and for consistency did that for the other applicable variables as well.

File: src/ws/main.c [hunk 2]
```diff
@@ -118,17 +118,16 @@ main (int argc,
       char *argv[])
 {
   gint ret = 1;
+  g_autoptr (GOptionContext) context = NULL;
```
Comment (allisonkarlitskaya) [inline src/ws/main.c]: No space here, please.

Comment (martinpitt) [reply src/ws/main.c]: Despaced!

**Comment by stefwalter** [comment]
I believe I have been overruled on g_autoptr().

**Pull request merged**